### PR TITLE
simplify domain and fix bugs

### PR DESCRIPTION
**Pull request by jonsterling**
it seemed like I needed this before, but now I don't.

### CODE REVIEW

File: src/core/Val.ml [hunk 1]
```diff
@@ -1208,6 +1201,9 @@ struct
             let phi = I.equate ri r'i in
             let phi0 = I.cmp (I.equate (`Atom x) `Dim0) phi in
             let yi, el = Abs.unleash absi in
+
+            (* If it helps, this right here is the source of the error that appears in omega1s1. - JMS *)
+            (* Format.eprintf "Here's a bad argument to rigid_vproj: %a@.@." pp_value el; *)
```
Comment (jonsterling) [inline src/core/Val.ml]: @favonia I don't know if this helps, but this is the call to `rigid_vproj` that causes the last example in omega1s1 to fail.

File: src/core/Val.ml [hunk 2]
```diff
@@ -1199,12 +1199,8 @@ struct
           let face =
             Face.map @@ fun ri r'i absi ->
             let phi = I.equate ri r'i in
-            let phi0 = I.cmp (I.equate (`Atom x) `Dim0) phi in
             let yi, el = Abs.unleash absi in
-
-            (* If it helps, this right here is the source of the error that appears in omega1s1. - JMS *)
-            (* Format.eprintf "Here's a bad argument to rigid_vproj: %a@.@." pp_value el; *)
-            Abs.bind yi @@ rigid_vproj x ~ty0:(Value.act phi0 ty0) ~ty1:(Value.act phi ty1) ~equiv:(Value.act phi0 equiv) ~el
+            Abs.bind yi @@ vproj phi (I.act phi @@ `Atom x) ~ty0:(fun phi -> Value.act phi ty0) ~ty1:ty1 ~equiv:(fun phi -> Value.act phi equiv) ~el
```
Comment (jonsterling) [inline src/core/Val.ml]: @favonia This change makes the `omega1s1` example go through, but I don't know if it is correct. Can you have a look?

Comment (favonia) [reply src/core/Val.ml]: So it is basically the same issue. Thanks!

Comment (favonia) [reply src/core/Val.ml]: How do you feel about this new continuation-based API?

Comment (jonsterling) [reply src/core/Val.ml]: I think it's alright, but I don't yet have the right intuitions for it. I think generally, diagonals in the current implementation are very fucked up.

**Comment by jonsterling** [comment]
@favonia By the way, I tried calculating that example with 5 instead of 2 and it took about a minute to compute.

**Comment by jonsterling** [comment]
One of the huge bottlenecks is actually just `open_var,close_var`, that stupid locally nameless crap that I needed in the elaborator and the unifier. Just to recall, the reason that stuff even exists is I found it too hard to write the unifier with de bruijn indices.

**Comment by favonia** [comment]
LOL

**Pull request merged**